### DISCUSSION TITLE
Incompatible operation of plugin Datarequest with version Ckan 2.9.5.

### DISCUSSION THREAD

**Discussion by muhammetcetinyamac** [ckan/ckan#6914] [Q&A] [answered]
Hello,

I noticed that plugin ckanext-datarequest works incompatible with version ckan 2.9.5.
Example errors are as follows
`  users = set([user['id'] for user in datarequest_dict['organization']['users']])
`
`KeyError: 'users'`

function name is different when checking captcha
should be `check_recaptcha` instead of `check_datarequest_recaptcha`

Will datarequest plugin be upgraded to Ckan 2.9.5? Or is there a problem with my codes?

I would be happy if you could help me.

**Comment by aminumoha** [comment, marked as answer]
Any solutions to this incompatibility yet?

**Reply by smotornyuk** [reply]
the CKAN core team does not maintain ckanext-datarequest, so it would have more sense to ask such questions inside ckanext-datarequest repo.

But I quickly looked at ckanext-datarequest documentation and am almost sure your code is wrong. You are using `datarequest_dict['organization']['users']`, but [documentation of ckanext-datarequest](https://github.com/conwetlab/ckanext-datarequests) has no mention of `organization` key inside the dictionary returned by API. You should never rely on undocumented items, even if you see them. 
I see that `organization_id` is explicitly mentioned among the data, returned by datarequest API. Try using it instead in order to fetch the organization and get its users.

As for the `check_recaptcha` function - I see no such function in ckanext-datarequest so you probably misunderstood something. CKAN has a `check_recaptcha` function, so you can use it. But note, CKAN never had `check_datarequest_recaptcha`, so if you have it in your code, you may need to adapt it before using CKAN's `check_recaptcha`

**Reply by muhammetcetinyamac** [reply]
> CKAN çekirdek ekibi ckanext-datarequest'i sürdürmez, bu nedenle bu tür soruları ckanext-datarequest deposunda sormak daha mantıklı olacaktır.
> 
> Ancak hemen ckanext-datarequest belgelerine baktım ve kodunuzun yanlış olduğundan neredeyse eminim. Kullanıyorsunuz `datarequest_dict['organization']['users']`, ancak [ckanext-datarequest belgelerinde](https://github.com/conwetlab/ckanext-datarequests)`organization` API tarafından döndürülen sözlükte anahtardan söz edilmiyor . Belgelenmemiş öğelere, görseniz bile asla güvenmemelisiniz. Datarequest API tarafından döndürülen veriler arasında açıkça belirtildiğini görüyorum . `organization_id`Kuruluşu getirmek ve kullanıcılarını almak için bunun yerine kullanmayı deneyin.
> 
> İşleve gelince `check_recaptcha`- ckanext-datarequest'te böyle bir işlev göremiyorum, bu yüzden muhtemelen bir şeyi yanlış anladınız. CKAN'ın bir `check_recaptcha`işlevi var, onu kullanabilirsiniz. Ancak, CKAN'da hiçbir zaman bulunmadığını unutmayın, `check_datarequest_recaptcha`bu nedenle kodunuzda varsa, CKAN'ları kullanmadan önce uyarlamanız gerekebilir.`check_recaptcha`

Thank you for answer
